research/picai/nnunet_scripts/transfer_train.py

```diff
@@ -0,0 +1,277 @@
+        "this number of processes is used for all configurations specified with -c. If it's a "
+        "list of numbers this list must have as many elements as there are configurations. We "
+        "then iterate over zip(configs, num_processes) to determine then umber of processes "
+        "used for each configuration. More processes is always faster (up to the number of "
+        "threads your PC can support, so 8 for a 4 core CPU with hyperthreading. If you don't "
+        "know what that is then dont touch it, or at least don't increase it!). DANGER: More "
+        "often than not the number of processes that can be used is limited by the amount of "
+        "RAM available. Image resampling takes up a lot of RAM. MONITOR RAM USAGE AND "
+        "DECREASE -np IF YOUR RAM FILLS UP TOO MUCH!. Default: 8 processes for 2d, 4 "
+        "for 3d_fullres, 8 for 3d_lowres and 4 for everything else",
+    )
+    parser.add_argument(
+        "--device",
+        type=str,
+        default="cuda",
+        required=False,
+        help="[OPTIONAL] Use this to set the device the training should run with. Available options are 'cuda' "
+        "(GPU), 'cpu' (CPU) and 'mps' (Apple M1/M2). Do NOT use this to set which GPU ID! "
+        "Use CUDA_VISIBLE_DEVICES=X nnUNetv2_train [...] instead!",
+    )
+    parser.add_argument(
+        "--skip_preprocessing",
+        action="store_true",
+        required=False,
+        help="[OPTIONAL] Use this to skip the preprocessing step if you have"
+        "already run this script with preprocessing for the given pretrained model"
+        "and finetuning dataset",
+    )
+    parser.add_argument(
+        "-tr",
+        type=str,
+        default="nnUNetTrainer",
+        required=False,
+        help="[OPTIONAL] The name of the nnunet trainer class to use. The default"
+        "is nnUNetTrainer but nnUNet provides alternatives such as nnUNetTrainer_1epoch",
+    )
+
+    args = parser.parse_args()
+
+    check_configs(id=args.pt_id, plans_identifier=args.pt_plans_identifier, configs=args.configs)
+
+    assert len(args.configs) == len(
+        args.pt_checkpoints
+    ), f"The number of configs ({len(args.configs)}) and number of pretrained \
+        checkpoints ({len(args.pt_checkpoints)}) do not match"
+
+    # Transfer Metadata needed for processing the finetuning set
+    tl_plans_identifier = transfer_metadata(
+        ft_id=args.ft_id, pt_id=args.pt_id, pt_plans_identifier=args.pt_plans_identifier
+    )
+
+    # Set default number of processes for preprocessing
+    if args.np is None:
+        default_np = {"2d": 8, "3d_fullres": 4, "3d_lowres": 8}
+        np = [default_np[c] if c in default_np.keys() else 4 for c in args.configs]
+    else:
+        np = args.np
+
+    # preprocessing
+    if not args.skip_preprocessing:
+        preprocess_dataset(dataset_id=args.ft_id, plans_identifier=tl_plans_identifier, num_processes=np)
```

Comment:
I think you are also able to pass the list of configurations to this function. If we don't than we may do preprocessing for an architecture that we don't care about training